src/main/java/org/truffleruby/core/string/StringNodes.java

```diff
@@ -4459,4 +4459,16 @@ private String formatTooLongError(int count, RubyString string) {
 
     }
 
+    @Primitive(name = "is_character_head", lowerFixnum = 2)
+    public abstract static class IsCharacterHeadPrimitiveNode extends PrimitiveArrayArgumentsNode {
+        @Child IsCharacterHeadNode isCharacterHeadNode = IsCharacterHeadNode.create();
```

Comment:
It's preferred to use `@Cached` than a `@Child` node (works better with the DSL and also more lazily allocated), could you switch to that?

Reply:
Sure